### PR TITLE
Cloud Credentials update

### PR DESCRIPTION
**Pull request by susesgartner**
## Issue: Update the cloud credentials to use V1 and fix how the credentials are loaded.
Shepherd PR: https://github.com/rancher/shepherd/pull/233
## Problem
The credentials were using v3 api which caused them not to get cleaned up and credentials were loaded in the creation function which is bad practice
 
## Solution
Made the function a bit more modular and updated it to use V1 credentials.
 
## Testing
Tested on all providers.

### CODE REVIEW

**Review by markusewalker — APPROVED** [review]
LGTM. Holding off on merging until shepherd PR is merged.

**Review by markusewalker — DISMISSED** [review]

**Review by Josh-Diamond — APPROVED** [review]

**Review by markusewalker — APPROVED** [review]

**Review by Josh-Diamond — APPROVED** [review]

**Pull request merged**
